plugins/jobtype/src/azkaban/jobtype/HadoopJavaJob.java

```diff
@@ -220,4 +230,23 @@ public String toString() {
         + _progressMethod + '\'' + ", _javaObject=" + _javaObject + ", props="
         + getJobProps() + '}';
   }
+
+  /**
+   * This cancel method, in addition to the default canceling behavior, also
+   * kills the MR jobs launched by this job on Hadoop
+   */
+  @Override
+  public void cancel() throws InterruptedException {
+    super.cancel();
+
+    info("Cancel called.  Killing the launched MR jobs on the cluster");
+
+    String azExecId = jobProps.getString("azkaban.flow.execid");
+    final String logFilePath =
+        String.format("%s/_job.%s.%s.log", getWorkingDirectory(), azExecId,
```

Comment:
"%s/_job.%s.%s.log" is used in all the cancel methods. Should we make it a constant for ?

It will be good to create getLogFilePath(workingDir, execId, getId) as it is common as well ?